src/hooks/useThemeProps/index.ts

```diff
@@ -0,0 +1,13 @@
+import {useContext, createContext} from 'react';
+import * as Modifiers from '../../commons/modifiers';
+import {ThemeManager} from 'style';
+
+const EmptyContext = createContext({});
+
+const useThemeProps = (props: any, componentName: string) => {
+  const themeContext = ThemeManager.getThemeContext();
+  const context = useContext(themeContext ?? EmptyContext);
+  return Modifiers.getThemeProps(props, context, componentName);
```

Comment:
I've just now noticed this is not memoized, it does create a new object and spreads props inside, WDYT?

Reply:
it's ok it's not memoize because of the spreading.. 
the inner props refs are kept, and that's what important